src/taskFunctions.js

```diff
@@ -0,0 +1,98 @@
+import chkboxStatus from './chkbox.js';
+
+class Task {
+  constructor(i, c = false, d) {
+    this.index = i;
+    this.completed = c;
+    this.description = d;
+  }
+}
+
+let taskList = [];
+
+const saveStorage = () => {
+  localStorage.setItem('tasks', JSON.stringify(taskList));
+};
+
+const addLi = (ul, index, completed, description) => {
+  const item = document.createElement('li');
+  const cBox = document.createElement('input');
+  cBox.type = 'checkbox';
+  cBox.id = index;
+  cBox.checked = completed;
+  cBox.addEventListener('change', () => {
+    if (chkboxStatus(cBox)) {
+      taskList[parseInt(cBox.id, 10)].completed = true;
+    } else {
+      taskList[parseInt(cBox.id, 10)].completed = false;
+    }
+    saveStorage();
+  });
+  const label = document.createElement('label');
+  label.contentEditable = true;
+  label.innerText = description;
+  label.addEventListener('focus', () => {
+    label.parentNode.style.backgroundColor = 'lightyellow';
+    label.parentNode.childNodes[2].style.pointerEvents = 'initial';
+    label.parentNode.childNodes[2].childNodes[0].style.display = 'none';
+    label.parentNode.childNodes[2].childNodes[2].style.display = 'initial';
+  });
+  label.addEventListener('focusout', (e) => {
+    label.parentNode.style.backgroundColor = 'initial';
+    label.parentNode.childNodes[2].childNodes[0].style.display = 'initial';
+    label.parentNode.childNodes[2].childNodes[2].style.display = 'none';
+    taskList[parseInt(cBox.id, 10)].description = e.composedPath()[0].innerText;
+    saveStorage();
+    setTimeout(() => {
+      label.parentNode.childNodes[2].style.pointerEvents = 'none';
+    }, 500);
+  });
+  const elBtn = document.createElement('button');
+  elBtn.className = 'element-button';
+  elBtn.addEventListener('click', (e) => {
+    // eslint-disable-next-line no-use-before-define
+    removeElement(ul, e.currentTarget.parentNode.childNodes[0].id);
+    e.target.closest('li').remove();
+  }, false);
+  const icon = document.createElement('i');
+  icon.className = 'fas fa-ellipsis-v';
+  const icon2 = document.createElement('i');
+  icon2.className = 'fas fa-trash';
+  elBtn.append(icon, icon2);
+  item.append(cBox, label, elBtn);
+  ul.prepend(item);
+};
+
+const refresh = (ul) => {
+  while (ul.hasChildNodes()) ul.removeChild(ul.firstChild);
```

Comment:
- [ ] it's recommended you use:
````
 ul.innerHTML = ""; 
````
to delete all the child nodes in ones, you will avoid to use that loop, and use less resources of the machine.

Reply:
I tried this approach at first, but after some reading is actually not recommended and might be a lot slower than you think. You can look it up because there is a lot of discussion about it, but one recent [post](https://rusingh.com/javascript-benchmark-removechild-vs-innerhtml/) actually tested the speed of these two approaches.